services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package kernelspace
+
+import (
+	"encoding/base64"
+	"net"
+
+	"github.com/mdlayher/wireguardctrl"
+	"github.com/mdlayher/wireguardctrl/wgtypes"
+	wg "github.com/mysteriumnetwork/node/services/wireguard"
+)
+
+var allowedIPs = []net.IPNet{
+	{IP: net.IPv4zero, Mask: net.CIDRMask(0, 32)},
+	{IP: net.IPv6zero, Mask: net.CIDRMask(0, 128)},
+}
+
+type client struct {
+	wgClient *wireguardctrl.Client
+}
+
+// NewWireguardClient creates new wireguard kernel space client.
+func NewWireguardClient() (*client, error) {
```

Comment:
Exported function returns unexported structure - something is wrong.

Reply:
This unexported struct implement interface and will not be used outside directly.
Replacing it to return the interface instead makes circular dependency.
Not sure if it really worth to change it.

Reply:
I am just saying that exported functions suppose to return exported structures. For example if someone will want to include your structure (wrap it) into another just because of the different interface - it won't be possible. If your function returns structure - that means structure is part of public api and should be exported - yes it can have all the fields private.